docs/website/docs/plus/ecosystem/delta.md

```diff
@@ -1,33 +1,191 @@
+```
+
+The Delta destination can also be defined in python as follows:
+
+```py
+pipeline = dlt.pipeline("loads_delta", destination="delta")
 ```
 
-You are still able to use regular filesystem configuration.
+
+## Write dispositions
+
+The Delta destination handles the write dispositions as follows:
+- `append` - files belonging to such tables are added to the dataset folder
+- `replace` - all files that belong to such tables are deleted from the dataset folder, and then the current set of files is added.
+- `merge` - can be used only with the `upsert` [merge strategy](../../general-usage/incremental-loading#upsert)
+
+:::caution
+The `upsert` merge strategy for the Delta destination is **experimental**.
+:::
+
+The `merge` write disposition can be configured as follows on the source/resource level:
+
+<Tabs values={[{"label": "dlt.yml", "value": "yaml"}, {"label": "Python", "value": "python"}]}  groupId="language" defaultValue="yaml">
+  <TabItem value="yaml">
+
+```yaml
+sources:
+  my_source:
+    type: sources.my_source
+    with_args:
+      write_disposition:
+        disposition: merge
+        strategy: upsert
+```
+  </TabItem>
+  <TabItem value="python">
+
 ```py
-from dlt_plus.destinations import delta
+@dlt.resource(
+    primary_key="id",  # merge_key also works; primary_key and merge_key may be used together
+    write_disposition={"disposition": "merge", "strategy": "upsert"},
+)
+def my_resource():
+    yield [
+        {"id": 1, "foo": "foo"},
+        {"id": 2, "foo": "bar"}
+    ]
+...
+
+pipeline = dlt.pipeline("loads_delta", destination="delta")
 
-dest_ = delta(destination_name="filesystem")
 ```
+</TabItem>
+</Tabs>
+
+Or on the `pipeline.run` level: <!-- can this also be defined in the yaml??-->
+
+```py
+pipeline.run(write_disposition={"disposition": "merge", "strategy": "upsert"})
+```
+
+## Partitioning
+
+Delta tables can be partitioned (using [Hive-style partitioning](https://delta.io/blog/pros-cons-hive-style-partionining/)) by specifying one or more partition column hints on the source/resource level:
+
+<Tabs values={[{"label": "dlt.yml", "value": "yaml"}, {"label": "Python", "value": "python"}]}  groupId="language" defaultValue="yaml">
+  <TabItem value="yaml">
+
+  ```yaml
+  sources:
+    my_source:
+      type: sources.my_source
+      with_args:
+        columns:
+          foo:
+            partition: True
+  ```
+
+  </TabItem>
+  <TabItem value="python">
+
+  ```py
+  @dlt.resource(
+    columns={"foo": {"partition": True}}
+  )
+  def my_resource():
+      ...
+
+  pipeline = dlt.pipeline("loads_delta", destination="delta")
+  ```
+
+  </TabItem>
+</Tabs>
+
+:::caution
+Partition evolution (changing partition columns after a table has been created) is not supported.
+:::
+
+## Table access helper functions
+You can use the `get_delta_tables` helper functions to access the native [DeltaTable](https://delta-io.github.io/delta-rs/api/delta_table/) objects. 
```

Comment:
Can you use the usual data access with `datasets`? If not, we should mention that